src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +1026,265 @@ def _n_to_index_(self, n):
+            sage: S
+            3-regular sequence 1, 3, 2, 3, 9, 6, 2, 6, 4, 3, ...
+            sage: from itertools import islice
+            sage: L = []; ps = 0
+            sage: for s in islice(S, 110):
+            ....:     ps += s
+            ....:     L.append(ps)
+            sage: G = Seq3.guess(lambda n: L[n])
+            sage: G
+            3-regular sequence 1, 4, 6, 9, 18, 24, 26, 32, 36, 39, ...
+            sage: G.mu[0], G.mu[1], G.mu[2], G.left, G.right
+            (
+            [ 0  1]  [18/5  2/5]  [ 6  0]
+            [-6  7], [18/5 27/5], [24  2], (1, 0), (1, 1)
+            )
+            sage: G == S.partial_sums(include_n=True)
+            True
+        """
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from sage.arith.srange import srange, xsrange
+        from sage.matrix.constructor import Matrix
+        from sage.misc.mrange import cantor_product
+        from sage.modules.free_module_element import vector
+
+        k = self.k
+        domain = self.coefficient_ring()
+        if sequence is None:
+            mu = [[] for _ in srange(k)]
+            seq = lambda m: tuple()
+        else:
+            mu = [M.rows() for M in sequence.mu]
+            seq = lambda m: sequence.left * sequence._mu_of_word_(
+                self._n_to_index_(m))
+
+        zero = domain(0)
+        one = domain(1)
+
+        def values(m, lines):
+            return tuple(seq(m)) + tuple(f(k**t_R * m + r_R) for t_R, r_R, s_R in lines)
+
+        @cached_function(key=lambda lines: len(lines))  # we assume that existing lines are not changed (we allow appending of new lines)
+        def some_inverse_U_matrix(lines):
+            d = len(seq(0)) + len(lines)
+
+            for m_indices in cantor_product(xsrange(n_max), repeat=d, min_slope=1):
+                U = Matrix(domain, d, d, [values(m, lines) for m in m_indices]).transpose()
+                try:
+                    return U.inverse(), m_indices
+                except ZeroDivisionError:
+                    pass
+            else:
+                raise RuntimeError
+
+        def guess_linear_dependence(t_L, r_L, lines):
+            iU, m_indices = some_inverse_U_matrix(lines)
+            X_L = vector(f(k**t_L * m + r_L) for m in m_indices)
+            return X_L * iU
+
+        def verify_linear_dependence(t_L, r_L, linear_dependence, lines):
+            return all(f(k**t_L * m + r_L) ==
+                       linear_dependence * vector(values(m, lines))
+                       for m in xsrange(0, (n_max - r_L) // k**t_L + 1))
+
+        def find_linear_dependence(t_L, r_L, lines):
+            linear_dependence = guess_linear_dependence(t_L, r_L, lines)
+            if not verify_linear_dependence(t_L, r_L, linear_dependence, lines):
+                raise ValueError
+            return linear_dependence
+
+        left = None
+        if seq(0):
```

Comment:
I do not understand the `if seq(0):` part: Why does it make a difference whether the zeroth element of the given sequence vanishes or not? I do not know whether it is related, but the following is not what I expect:

```python
sage: def s(k):
sage:     return k
sage: Seq2 = kRegularSequenceSpace(2, ZZ)
sage: S1 = Seq2.guess(s)
sage: S2 = Seq2.guess(s, sequence=S1)
sage: S1
2-regular sequence 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, ...
sage: S2
2-regular sequence 0, -2, -2, -6, -2, -6, -10, -14, -2, -6, ...
```
https://github.com/sagemath/sage/issues/21204#issuecomment-1417820841 Item 12

Reply:
`seq(0)` is not the zeroth element of the given sequence, but the vector-valued sequence (`left`) when reading `0`, so the `if` asks whether we have the zero-vector or not. This is not very nicely written, so rewritten in 9fc2663749b.

Reply:
Your example fails due to the left/right issue discussed in the other conversations. It works now. I've added your example as 250a98c2e58.

The current code does only drop the bootstrapping sequence, if this is a zero-sequence (due to left vector 0), which should be fine to do.